src/bucket.rs

```diff
@@ -311,6 +318,7 @@ impl Display for BucketError {
         match self {
             Self::UnsupportedScheme => f.write_str("unsupported Url scheme"),
             Self::MissingHost => f.write_str("Url is missing the `host`"),
+            Self::ParseError(e) => f.write_str(&e.to_string()),
```

Comment:
Something like this should be a bit nicer and avoid an heap allocation

```suggestion
            Self::ParseError(e) => e.fmt(f),
```